src/renderer/src/lib.rs

```diff
@@ -1,16 +1,151 @@
 #![crate_name = "amethyst_renderer"]
 #![crate_type = "lib"]
 #![doc(html_logo_url = "http://tinyurl.com/hgsb45k")]
+#![allow(dead_code)]
 
 //! High-level rendering engine with multiple backends.
 
+#[macro_use]
 extern crate gfx;
+#[macro_use]
+extern crate mopa;
 
-mod backend;
-mod frame;
-mod frontend;
-mod pipeline;
+extern crate glutin;
+extern crate cgmath;
 
-pub use self::frame::{Frame, FrameBuilder};
-pub use self::frontend::Renderer;
-pub use self::pipeline::{Pipeline, PipelineBuilder, Stage, Step};
+pub mod framebuffer;
+pub mod pass;
+pub mod method;
+
+use std::any::TypeId;
+use std::collections::HashMap;
+
+pub use pass::Pass;
+pub use framebuffer::Framebuffer;
+pub use method::Method;
+
+pub struct Renderer<R: gfx::Resources, C: gfx::CommandBuffer<R>> {
+    command_buffer: gfx::Encoder<R, C>,
+    methods: HashMap<(TypeId, TypeId), Box<Fn(&Box<Pass>, &Framebuffer, &Frame<R>, &mut gfx::Encoder<R, C>)>>
```

Comment:
I wonder if making a trait instead of this function would allow you to remove the `Pass` boxing, thus stripping the need to derive from `Any` for passes.